spec/models/generic_event/move_start_spec.rb

```diff
@@ -1,18 +1,140 @@
+    end
+  end
+
+  shared_examples_for 'a per confirmation webhook is not sent' do
+    before { allow(Notifier).to receive(:prepare_notifications) }
 
-  describe '#trigger' do
-    it 'does not persist changes to the eventable' do
-      generic_event.trigger
-      expect(generic_event.eventable).not_to be_persisted
+    it do
+      move_start_event.trigger
+      expect(Notifier).not_to have_received(:prepare_notifications).with(topic: person_escort_record, action_name: 'confirm_person_escort_record')
     end
+  end
+
+  let(:move) { move_start_event.eventable }
+
+  context 'when the move status is requested' do
+    let(:move_status) { :requested }
+
+    it_behaves_like 'the move status does not change'
+  end
+
+  context 'when the move status is booked' do
+    let(:move_status) { :booked }
+
+    it_behaves_like 'the move changes but is not saved'
+    it_behaves_like 'the move status changes to in_transit'
+
+    context 'when the move has an associated PER' do
+      subject(:move_start_event) { build(:event_move_start, eventable: create(:move, move_status, :with_person_escort_record, person_escort_record_status: person_escort_record_status)) }
+
+      let(:person_escort_record) { move.person_escort_record }
+
+      context 'when the PER status is unstarted' do
+        let(:person_escort_record_status) { 'unstarted' }
+
+        it_behaves_like 'the move status changes to in_transit'
+        it_behaves_like 'the PER status does not change'
+        it_behaves_like 'a GenericEvent::PerConfirmation event is not created'
+        it_behaves_like 'a per confirmation webhook is not sent'
+      end
+
+      context 'when the PER status is in_progress' do
+        let(:person_escort_record_status) { 'in_progress' }
+
+        it_behaves_like 'the move status changes to in_transit'
+        it_behaves_like 'the PER status does not change'
+        it_behaves_like 'a GenericEvent::PerConfirmation event is not created'
+        it_behaves_like 'a per confirmation webhook is not sent'
+      end
 
-    it 'sets the eventable `status` to in_transit' do
-      expect { generic_event.trigger }.to change { generic_event.eventable.status }.from('booked').to('in_transit')
+      context 'when the PER status is completed' do
+        let(:person_escort_record_status) { 'completed' }
+
+        it_behaves_like 'the move status changes to in_transit'
+        it_behaves_like 'the PER status changes to confirmed'
+        it_behaves_like 'a GenericEvent::PerConfirmation event is created'
+        it_behaves_like 'a per confirmation webhook is sent'
+      end
+
+      context 'when the PER status is confirmed' do
+        let(:person_escort_record_status) { 'confirmed' }
+
+        it_behaves_like 'the move status changes to in_transit'
+        it_behaves_like 'the PER status does not change'
+        it_behaves_like 'a GenericEvent::PerConfirmation event is not created'
```

Comment:
Just so I know I understand it - we don't expect an event here because the confirmation has happened in the past (and so the event would've been created) and the change to the move should not re-issue that event?

Reply:
yep, that's correct - it would have been handed over and instead would have a PerHandover event